Assets/MRTK/Tests/PlayModeTests/SlateTests.cs

```diff
@@ -222,6 +225,206 @@ public IEnumerator InstantiateGGVScroll()
+
+            Vector3 screenPoint = CameraCache.Main.ViewportToScreenPoint(new Vector3(0.5f, 0.0f, 0.5f));
+            for (var i = 0; i < 3; i++)
+            {
+                yield return handRight.MoveTo(CameraCache.Main.ScreenToWorldPoint(screenPoint));
+                yield return handRight.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+                yield return handRight.Move(new Vector3(0, 0.5f, 0), 10);
+                yield return handRight.SetGesture(ArticulatedHandPose.GestureId.Open);
+            }
+
+            var uvs = new List<Vector2>();
+            meshFilter.mesh.GetUVs(0, uvs);
+
+            Assert.AreEqual(-maxPan, uvs[0].y, 0.05, "mesh uv is not correct");
+
+            yield return handRight.Hide();
+        }
+
+        /// <summary>
+        /// Test scroll left pan limit instantiated from prefab
+        /// </summary>
+        [UnityTest]
+        public IEnumerator PrefabScrollLeftLimited()
+        {
+            var maxPan = 1.2f;
+            InstantiatePanLimitedSlateFromPrefab(maxPanHorizontal: maxPan, maxPanVertical: maxPan);
+
+            TestHand handRight = new TestHand(Handedness.Right);
+            yield return handRight.Show(Vector3.zero);
+
+            Vector3 screenPoint = CameraCache.Main.ViewportToScreenPoint(new Vector3(0.5f, 0.0f, 0.5f));
+            for (var i = 0; i < 3; i++)
+            {
+                yield return handRight.MoveTo(CameraCache.Main.ScreenToWorldPoint(screenPoint));
+                yield return handRight.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+                yield return handRight.Move(new Vector3(0.5f, 0f, 0), 10);
+                yield return handRight.SetGesture(ArticulatedHandPose.GestureId.Open);
+            }
+
+            var uvs = new List<Vector2>();
+            meshFilter.mesh.GetUVs(0, uvs);
+            Assert.AreEqual(-maxPan, uvs[0].x, 0.05, "mesh uv is not correct");
+
+            yield return handRight.Hide();
+        }
+
+        /// <summary>
+        /// Test scroll right pan limit instantiated from prefab
+        /// </summary>
+        [UnityTest]
+        public IEnumerator PrefabScrollRightLimited()
+        {
+            var maxPan = 1.2f;
+            InstantiatePanLimitedSlateFromPrefab(maxPanHorizontal: maxPan, maxPanVertical: maxPan);
+
+            TestHand handRight = new TestHand(Handedness.Right);
+            yield return handRight.Show(Vector3.zero);
+
+            Vector3 screenPoint = CameraCache.Main.ViewportToScreenPoint(new Vector3(0.5f, 0.0f, 0.5f));
+            for (var i = 0; i < 3; i++)
+            {
+                yield return handRight.MoveTo(CameraCache.Main.ScreenToWorldPoint(screenPoint));
+                yield return handRight.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+                yield return handRight.Move(new Vector3(-0.5f, 0f, 0), 10);
+                yield return handRight.SetGesture(ArticulatedHandPose.GestureId.Open);
+            }
+
```

Comment:
PrefabScrollRightLimited, PrefabScrollLeftLimited, PrefabScrollUpLimited and PrefabScrollDownLimited have very similar code. Could you please combine them or put duplicated code into a method?

Reply:
I've created ScrollToLimit method and remove duplicated code.